code/modules/events/holiday/xmas.dm

```diff
@@ -109,58 +109,57 @@
-				santa = new /mob/living/carbon/human(pick(GLOB.blobstart))
-				santa.key = M.key
-				qdel(M)
-
-				santa.real_name = "Santa Claus"
-				santa.name = "Santa Claus"
-				santa.mind.name = "Santa Claus"
-				santa.mind.assigned_role = "Santa"
-				santa.mind.special_role = "Santa"
-
-				santa.hair_style = "Long Hair"
-				santa.facial_hair_style = "Full Beard"
-				santa.hair_color = "FFF"
-				santa.facial_hair_color = "FFF"
-
-				santa.equip_to_slot_or_del(new /obj/item/clothing/under/color/red, slot_w_uniform)
-				santa.equip_to_slot_or_del(new /obj/item/clothing/suit/space/santa, slot_wear_suit)
-				santa.equip_to_slot_or_del(new /obj/item/clothing/head/santa, slot_head)
-				santa.equip_to_slot_or_del(new /obj/item/clothing/mask/breath, slot_wear_mask)
-				santa.equip_to_slot_or_del(new /obj/item/clothing/gloves/color/red, slot_gloves)
-				santa.equip_to_slot_or_del(new /obj/item/clothing/shoes/sneakers/red, slot_shoes)
-				santa.equip_to_slot_or_del(new /obj/item/tank/internals/emergency_oxygen/double, slot_belt)
-				santa.equip_to_slot_or_del(new /obj/item/device/radio/headset/heads/captain, slot_ears)
-				santa.equip_to_slot_or_del(new /obj/item/storage/backpack/santabag, slot_back)
-				santa.equip_to_slot_or_del(new /obj/item/device/flashlight, slot_r_store) //most blob spawn locations are really dark.
-
-				var/obj/item/card/id/gold/santacard = new(santa)
-				santacard.update_label("Santa Claus", "Santa")
-				var/datum/job/captain/J = new/datum/job/captain
-				santacard.access = J.get_access()
-				santa.equip_to_slot_or_del(santacard, slot_wear_id)
-
-				santa.update_icons()
-
-				var/obj/item/storage/backpack/bag = santa.back
-				var/obj/item/a_gift/gift = new(santa)
-				while(bag.can_be_inserted(gift, 1))
-					bag.handle_item_insertion(gift, 1)
-					gift = new(santa)
-
-				var/datum/objective/santa_objective = new()
-				santa_objective.explanation_text = "Bring joy and presents to the station!"
-				santa_objective.completed = 1 //lets cut our santas some slack.
-				santa_objective.owner = santa.mind
-				santa.mind.objectives += santa_objective
-				santa.mind.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/conjure/presents)
-				var/obj/effect/proc_holder/spell/targeted/area_teleport/teleport/telespell = new(santa)
-				telespell.clothes_req = 0 //santa robes aren't actually magical.
-				santa.mind.AddSpell(telespell) //does the station have chimneys? WHO KNOWS!
-
-				to_chat(santa, "<span class='boldannounce'>You are Santa! Your objective is to bring joy to the people on this station. You can conjure more presents using a spell, and there are several presents in your bag.</span>")
+	var/list/candidates = pollGhostCandidates("Santa is coming to town! Do you want to be santa?", poll_time=150)
+	if(LAZYLEN(candidates))
+		var/mob/dead/observer/Z = pick(candidates)
+		santa = new /mob/living/carbon/human(pick(GLOB.blobstart))
+		santa.key = Z.key
```

Comment:
I'd HIGHLY recommend just making the santa thing a datum outfit and having the person spawn in as a randomized human and then equipping it because this... just screams something wrong with me.

Reply:
out of scope, this PR is just making the event poll ghosts properly

Reply:
its code improvement and code improvement you would be in scope, especially because your already touching the file.

Reply:
it's not in scope, dont' backseat maintain

Reply:
oh you bullied them into doing it anyway